control-plane/subcommand/inject-connect/command.go

```diff
@@ -420,6 +423,10 @@ func (c *Command) validateFlags() error {
 		return errors.New("-consul-dataplane-image must be set")
 	}
 
+	if c.flagV2Tenancy && !c.flagResourceAPIs {
+		return errors.New("-enable-resource-apis must be set to 'true' if -enable-v2tenancy is set")
+	}
```

Comment:
Looking at this and some recent info shared around core changes, I think I now see why this makes sense. Still wondering whether we need a separate flag in K8s - at least right now, given this check - but I follow the rationale for splitting in Core.

Reply:
Good question - this is a bit nuanced and a result of the split brain we introduced in Consul 1.17 where v2 catalog and mesh resource are backed by v1 tenancy and acls. 

This allowed multiport to ship in 1.17 in hybrid mode. With consul 1.18, we need to be explicit about whether we're in "hybrid" mode (v2 catalog + v2 mesh + v1 tenancy + v1 acls) or "full v2" mode (v2 catalog + v2 mesh + v2 tenancy + ACLs not supported).

As a result, all affected components need to be aware and flow accordingly:
- resources-apis=true => hybrid mode
- resource-apis=true and v2tenancy=true => full v2 mode and no ACLs

In short, this is how the `connect-inject` subcommand knows whether to spin up the v1 namespace controller or the v2 namespace controller. Never both.

Reply:
That makes sense, thanks for the extra context @analogue ! If that's the established pattern then it makes sense to keep it consistent here.

~ Small request: could we add a blurb comment along those lines here or somewhere else appropriate so that context is visible?

Reply:
Updated in https://github.com/hashicorp/consul-k8s/pull/3435/commits/fa47ad03964628c984ded0f5e69d5afb8f37393c

Reply:
Thank you! 🎨